File: src/main/java/github/buried_in_code/console/Colour.java
package github.buried_in_code.console;

import org.jetbrains.annotations.NotNull;

/**
 * Created by BuriedInCode on 2021-Jun-02.
 */
public enum Colour {
	RESET("\u001B[0m"),
	BLACK("\u001B[30m"),
	RED("\u001B[31m"),
	GREEN("\u001B[32m"),
	YELLOW("\u001B[33m"),
	BLUE("\u001B[34m"),
	MAGENTA("\u001B[35m"),
	CYAN("\u001B[36m"),
	WHITE("\u001B[37m");

	@NotNull
	private final String ansicode;

	Colour(@NotNull String ansicode) {
		this.ansicode = ansicode;
	}

	@NotNull
	@Override
	public String toString() {
		return ansicode;
	}
}

File: Base/src/main/kotlin/macro303/base/Colour.kt
package macro303.base

import java.awt.Color

enum class Colour(val colourCode: String, private val colour: Color) {
	RESET("\u001B[0m", Color.BLACK),
	RED("\u001B[31m", Color.RED),
	GREEN("\u001B[32m", Color.GREEN),
	YELLOW("\u001B[33m", Color.YELLOW),
	BLUE("\u001B[34m", Color.BLUE),
	MAGENTA("\u001B[35m", Color.MAGENTA),
	CYAN("\u001B[36m", Color.CYAN),
	WHITE("\u001B[37m", Color.WHITE);

	val hexCode: String
		get() = String.format("#%02X%02X%02X%02X", colour.red, colour.green, colour.blue, colour.alpha)
}